core/rawdb/database.go

```diff
@@ -218,19 +218,17 @@ func NewDatabaseWithFreezer(db ethdb.KeyValueStore, freezer string, namespace st
 			// Otherwise, key-value store continues where the freezer left off, all is fine.
 			// We might have duplicate blocks (crash after freezer write but before key-value
 			// store deletion, but that's fine).
-		} else {
+		} else if ReadHeadHeaderHash(db) != common.BytesToHash(kvgenesis) {
```

Comment:
Please revert this. The reason these were split is because it's easier to follow with the comments embedded at the correct positions.

Reply:
Thanks for your advice, resolved.